index.js

```diff
@@ -38,7 +42,10 @@ Notifier.prototype.handler = function( request, response ) {
 			}
 			data = JSON.parse( data );
 		} catch( error ) {
+
 			// Invalid data, stop processing
+			error = new Error( "Invalid data: " + error.message );
+			error.payload = data;
```

Comment:
At this point, `data` is already parsed as a query string. Let's change `data` to `rawData` and then have `var data = rawData` as the first line of the `end` event.